pkg/actors/actors.go

```diff
@@ -85,14 +85,22 @@ type Actors interface {
 	GetActiveActorsCount(ctx context.Context) []ActiveActorsCount
 }
 
+// PlacementService allows for interacting with the actor placement service.
+type PlacementService interface {
+	Start()
+	Stop()
+	WaitUntilPlacementTableIsReady(ctx context.Context) error
+	LookupActor(actorType, actorID string) (host string, appID string)
+}
+
 // GRPCConnectionFn is the type of the function that returns a gRPC connection
 type GRPCConnectionFn func(ctx context.Context, address string, id string, namespace string, customOpts ...grpc.DialOption) (*grpc.ClientConn, func(destroy bool), error)
 
 type actorsRuntime struct {
-	appChannel             channel.AppChannel
+	externalAppChannel     channel.AppChannel
```

Comment:
I feel that the names "external" and "internal" don't fit well with what we want to implement for actors. The Actor being internal doesn't mean we have an external app, does it? it seems to me that we need an ActorsChannel (which embeds an AppChannel) maybe? And it is able to decide whether to call an internal actor or an app actor. I don't know if the concept of "external" is the best one to use, but this is just an impression of someone who is reviewing a workflow PR for the first time.

I would avoid to create a new concept of "external app" because of the internal actors - which is kinda what is happening here.

Reply:
I don't think I'm understanding the nuance of your feedback, but I'll try to respond. "External" in this context means external to the sidecar (same with "Internal"). The reason to rename `appChannel` to `externalAppChannel` is simply to distinguish it from the newly added `internalAppChannel`. If "external" is confusing, I can certainly remove it, or perhaps name it `defaultAppChannel`?

Reply:
> Or, perhaps what I can do is rename "internalAppChannel" to "internalActorChannel" and revert "externalAppChannel" back to "appChannel"

That sounds better. Just to clarify: My concern is around the term"external/internal" for apps. There's no mention to an "external" app or "internal" app in our documentation, right? The usage of Internal actors doesn't mean that we have an internal app, does it ? You don't need to tie these concepts, we can have an internal actor without creating a concept of "internal app".

Reply:
Correct, none of this is in our documentation because these are not concepts that are exposed to users. Rather, this is something that’s only relevant to Dapr maintainers.

We will mention that there is a thing called “internal actors” in the workflow operational docs, but only as a way to help explain the dependencies that are involved when running workflows.

I went ahead and made these naming changes in the previous commit. I appreciate the feedback and think that this will help reduce conceptual overhead for maintainers.